File: authorization-admin-service-spring/src/main/java/eu/rudisch/authorization/admin/repository/UserRepository.java
package eu.rudisch.authorization.admin.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import eu.rudisch.authorization.admin.model.User;

public interface UserRepository extends JpaRepository<User, Integer> {
	Optional<User> findByUsername(String name);
}

File: authorization-admin-service-spring-kotlin/src/main/kotlin/eu/rudisch/authorization/admin/repository/UserRepository.kt
package eu.rudisch.authorization.admin.repository

import eu.rudisch.authorization.admin.model.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserRepository : JpaRepository<User, Int> {
	fun findByUsername(name: String): Optional<User>
}